specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,438 @@
+{
+    auto channelUri = result.Channel().Uri();
+    auto channelExpiry = result.Channel().ExpirationTime();
+
+    // Persist the channelUri and Expiry in the App Service
+}
+else if (result.Status() == ChannelStatus::CompletedFailure)
+{
+    LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with channel request!");
+}
+```
+
+## Manifest Registration
+For MSIX, the COM activator GUID and the exe path need to be registered in the manifest. The launch args would need to be pre set to a well-known string that defines Push Triggers. 
+```xml
+<Extensions>
+  <com:Extension Category="windows.comServer">
+    <com:ComServer>
+      <com:ExeServer Executable="SampleBackgroundApp\SampleBackgroundApp.exe" DisplayName="SampleBackgroundApp" Arguments="-ReunionPushServer">
+        <com:Class Id="{GUIDEntryPoint}" DisplayName="Reunion Push" />
+      </com:ExeServer>
+    </com:ComServer>
+  </com:Extension>
+</Extensions>
+```
+
+## UWP InProc activator
+The UWP inproc activator flow differs from the Win32 inproc activator flow since there is technically no WinMain activation associated with UWPs. Instead the App will have to  override and implement OnBackgroundActivated which is tightly coupled to Windows.UI.Xaml workstreams.
+
+# API Details
+```c# (but really MIDL3)
+    namespace Microsoft.ProjectReunion
+    {
+        // The legacy ActivationKind does not support operations like Push. So we need to define an extension set
+        enum ActivationKindExtension
+        {
+            Launch = 0,
+            Search = 1,
+            ShareTarget = 2,
+            File = 3,
+            Protocol = 4,
+            FileOpenPicker = 5,
+            FileSavePicker = 6,
+            CachedFileUpdater = 7,
+            ContactPicker = 8,
+            Device = 9,
+            PrintTaskSettings = 10,
+            CameraSettings = 11,
+            RestrictedLaunch = 12,
+            AppointmentsProvider = 13,
+            Contact = 14,
+            LockScreenCall = 15,
+            VoiceCommand = 16,
+            LockScreen = 17,
+            PickerReturned = 1000,
+            WalletAction = 1001,
+            PickFileContinuation = 1002,
+            PickSaveFileContinuation = 1003,
+            PickFolderContinuation = 1004,
+            WebAuthenticationBrokerContinuation = 1005,
+            WebAccountProvider = 1006,
+            ComponentUI = 1007,
+            ProtocolForResults = 1009,
+            ToastNotification = 1010,
+            Print3DWorkflow = 1011,
+            DialReceiver = 1012,
+            DevicePairing = 1013,
+            UserDataAccountsProvider = 1014,
+            FilePickerExperience = 1015,
+            LockScreenComponent = 1016,
+            ContactPanel = 1017,
+            PrintWorkflowForegroundTask = 1018,
+            GameUIProvider = 1019,
+            StartupTask = 1020,
+            CommandLineLaunch = 1021,
+            BarcodeScannerProvider = 1022,
+            Push = 1023,
```

Comment:
Its unfortunate that we need to resort to this.  Ongoing discussions to introduce New Kind extension support that isn't part of Foundations today.